### ISSUE TITLE
Unstable training loss

### ISSUE THREAD

**Issue by cedricbulteel** [vikhyat/moondream#304]
When executing the `finetune_region.py`script, the training loss behaves very unstable and doesn't decrease. The image shows the loss for training on my custom dataset. It's not a dataset issue as this behavior also appeared for the default WasteDetection dataset

<img width="346" height="300" alt="Image" src="https://github.com/user-attachments/assets/c15326dc-dda6-446d-8d16-103f5d046f22" />
